tck_test.go

```diff
@@ -0,0 +1,59 @@
+package httpwasm
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/http-wasm/http-wasm-host-go/tck"
+	"github.com/labstack/echo/v4"
+)
+
+// BackendHandler is a http.Handler implementing the logic expected by the TCK.
+// It serves to echo back information from the request to the response for
+// checking expectations.
+func backendHandler() echo.HandlerFunc {
+	return func(c echo.Context) error {
+		c.Response().Header().Set("x-httpwasm-next-method", c.Request().Method)
+		c.Response().Header().Set("x-httpwasm-next-uri", c.Request().RequestURI)
+		for k, vs := range c.Request().Header {
+			for i, v := range vs {
+				c.Response().Header().Add(fmt.Sprintf("x-httpwasm-next-header-%s-%d", k, i), v)
+			}
+		}
+		return nil
+	}
+}
+
+const testPort = 1323
+
+func TestTCK(t *testing.T) {
```

Comment:
kudos to @anuraaga as I could easily port the tests here. As a side note, I would include the feature of finding a free port for http servers in the tck as httptest does.